monai/auto3dseg/algo_gen.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from monai.transforms import Randomizable
+
+
+class Algo:
+    """
+    An algorithm in this context is loosely defined as a data processing pipeline consisting of multiple components
+    such as image preprocessing, followed by deep learning model training and evaluation.
+    """
+
+    def set_data_stats(self, *args, **kwargs):
+        """Provide dataset (and summaries) so that the model creation can depend on the input datasets."""
+        pass
+
+    def train(self, params: dict):
+        """
+        Read training/validation data and output a model.
+
+        Args:
+            params: key-value pairs of input parameters for the training pipeline.
+        """
+        pass
+
+    def predict(self, params: dict):
+        """
+        Read test data and output model predictions.
+
+        Args:
+            params: key-value pairs of input parameters for the predicting pipeline.
+        """
+        pass
+
+    def get_score(self, *args, **kwargs):
+        """Returns the model quality measurement based on training and validation datasets."""
+        pass
+
+
+class AlgoGen(Randomizable):
```

Comment:
why the algogen is randomizable?

Reply:
so that it's possible to draw random samples from a probabilistic search space